compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2713,6 +2714,42 @@ private void checkConstantAssignment(BLangExpression varRef, AnalyzerData data)
         }
     }
 
+    private boolean isFiniteWithNilMember(BType bType) {
```

Comment:
Is this name accurate? The check is for a finite type with all `null` members, right? Right now it kinda implies the check is for a finite type with at least one `null` member?

Let's use `null` here because it's specifically the null literal.